### ISSUE TITLE
drop lazyLoad feature from core

### ISSUE THREAD

**Issue by graingert** [pablojim/highcharts-ng#486]
The modern approach for webapps is to create bundles using browserify, webpack, rollup, or even just requirejs+almond. Those users who use a bundler (now the recommend way to create angular apps) do not need to lazyLoad their packages. The lazyLoading approach should also be discouraged.

Instead I think it would be appropriate to create an over-ridable highchartsLib service, and separate highcharts-ng/lazyload that implements it.

```js
// highcharts-ng
function highchartsLib($q, $window) {
    return $q.resolved($window.Highcarts); 
}

function highcharts(){
   return {
       ...
       // directive config
       controller: function (highchartsLib) {
       ...
        highchartsLib.then(Highcharts => processChartConfig(Highcharts));
      }
    }
}

angular.module('highcharts-ng').factory('highchartsLib', ['$q', '$window', '$timeout'], highchartsLib]).directive(highcharts);
```

```js
// highcharts-ng-lazyload
function highchartsLib($q, $window) {
    return $q(resolve => magicLazyLoadStuff(() => resolve($window.Highcarts))); 
}
angular.module('highcharts-ng-lazyload', ['highcharts-ng']).factory('highchartsLib', highchartsLib);
```

Then people who really need the lazyLoading functionality can do:

bower install highcharts-ng
```
<script src="bower_components/highcharts-ng/index.js"></script>
<script src="bower_components/highcharts-ng/lazyload.js"></script>
<script>angular.module('some-app-name', ['highcharts-ng-lazyload']).component(...)</script>
```

**Comment by pablojim** [comment]
How about a slight different approach? Keep it bundled in the same bower dependency but factor it out as suggested. That way there is still one repo/artifact to maintain. The overhead for users who don't use the lazyloading feature should be minimal.

**Comment by graingert** [comment]
@pablojim ok keep it in the same repo, I'll update my proposal

**Comment by graingert** [comment]
@pablojim having lazyload as a separate js file will keep it out of module bundlers

**Comment by pablojim** [comment]
Great Thanks @graingert - pull requests accepted!

**Comment by pablojim** [comment]
closed by #493

**Comment by graingert** [comment]
@pablojim can you reopen it's only half done?
